### ISSUE TITLE
so3_Exp autograd doesn't evaluate to numerical gradient

### ISSUE THREAD

**Issue by gmedan** [pypose/pypose#335]
### 🐛 Describe the bug

Although not very useful, the function `x.Exp().sum()` where `x.ltype == pypose.lietensor.lietensor.so3Type` should have an autograd result that is very close to a numerical gradient calculation. However, the following test fails in pypose 0.6.7:

```
def test_so3_Exp_grad():
    import torch
    import pypose as pp
    
    x = pp.randn_so3((), sigma=2.e-2,requires_grad=True, dtype=torch.double)

    delta = torch.randn(x.shape, dtype=x.dtype) * 1e-8
    f = lambda x: x.Exp().sum()
    # f = lambda x: x.sum()
    loss = f(x)
    loss.backward()
    analytical_grad = x.grad.dot(delta/delta.norm())

    loss_plus = f(pp.so3(x.tensor()+delta))
    loss_minus = f(pp.so3(x.tensor()-delta))
    numerical_grad = (loss_plus-loss_minus)/delta.norm()/2.0

    torch.testing.assert_close((analytical_grad/numerical_grad).item(), 1.0, rtol=1e-7, atol=1e-7)

```

### Versions

```
Collecting environment information...
PyTorch version: 2.1.2+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.10.13 (main, Aug 25 2023, 13:20:03) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-6.5.0-15-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 545.23.08
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.8.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.8.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.2
[pip3] pytorch-ignite==0.4.11
[pip3] pytorch3d==0.7.5
[pip3] torch==2.1.2
[pip3] torchvision==0.16.2
[conda] Could not collect
```

**Comment by pyposebot** [comment]
Hi, the definition of gradient/jacobian of Lie group is different from Euclidean space. Gradient of Lie Group is:
<img width="609" alt="image" src="https://github.com/pypose/pypose/assets/109780734/dcce320b-2e8f-444a-b6f2-0baf616b1d89">
More details can be also found in this [pypose.add doc](https://pypose.org/docs/main/generated/pypose.add/?highlight=add#pypose.add).
